File: checkIO/Elementary/secret-message.go
package main

import (
    "fmt"
    "regexp"
)

func findMessage(data string) {
    re := regexp.MustCompile("[^A-Z]")
    fmt.Println(re.ReplaceAllString(data, ""))
}

func main() {
    findMessage("How are you? Eh, ok. Low or Lower? Ohhh.") //"HELLO", "hello"
    findMessage("hello world!") //"", "Nothing"
    findMessage("HELLO WORLD!!!")   //"HELLOWORLD", "Capitals"
}

/*
function findMessage(data) {
    var pars = '';
	for(var i = 0; i < data.length; i++){
		if(data[i] != data[i].toLowerCase()){
			pars += data[i];
		};
	};
	if(pars == "HELLO"){
		return pars;
	}else if(!pars){
		return pars;
	}
	return pars;
};
*/

/*
var findMessage = data => data.replace(/[^A-Z]/g, '');
*/

File: CheckIO_py/secret-message.py
import re

def find_message(text):
    """Find a secret message"""
    p = re.compile('[A-Z]')
    m = p.findall(text)

    ret = ""
    for v in m:
        ret += v
    #for v in text:
        #if v == "x":
            #ret += v
    #"S".isupper() == true
    return ret

if __name__ == '__main__':
    #These "asserts" using only for self-checking and not necessary for auto-testing
    r = find_message("How are you? Eh, ok. Low or Lower? Ohhh.")
    print(r)
    #assert find_message("hello world!") == "", "Nothing"
    #assert find_message("HELLO WORLD!!!") == "HELLOWORLD", "Capitals"
